sources/minishell.h

```diff
@@ -20,6 +20,7 @@
 # include <readline/history.h>
 # include "../libft/libft.h"
 # include "resources.h"
+# include "commands/commands.h"
```

Comment:
1. Every minishell C file should include minishell.h. It's the main header for the project. Pretty much every function in minishell will use param structure, which is defined in minishell.h, so this is the simplest method here, include minishell.h from everywhere.
2. Every libft C file should include libft.h. Libft C files do NOT include minishell.h for portability.
3. minishell.h includes libft and all external headers (stdio etc)
4. if just one file X in minishell needs a specific external function, you can include it right in X. Or you can include this function in minishell.h, I don't mind here. There is a chance we'd have to expand this file X to two files, so maybe having a transitive dependency through minishell.h could be less bothersome.
5. I want to split definitions for minishell.h into several files, so that we don't have to read 300 lines every time. So I created some subheaders, like resources, commands etc.
6. Subheaders include minishell.h because they need param definition. minishell.h includes all subheaders because this way it provides a transitive dependency. If a file needs commands, it doesn't have to import commands.h, it can import minishell.h for params, and get commands.h as an subincluded bonus.
7. There is no recursive conflict here, because all headers are protected with ifdef HEADER_H. Here's a scenario:
- option_env.c includes minishell.h. So compiler knows that it has to copy-paste all contents of minishell.h into the beginning of option_env.c
- compiler starts to collect minishell.h contents. It notices that minishell.h includes commands.h, so contents of commands.h should be copy-pasted into the beginning of minishell.h
- compiler starts to collect commands.h contents. It notices that it includes minishell.h, so contents of minishell.h should be copy-pasted into the beginning of commands.h. BUT by that time the compiler has already defined MINISHELL_H variable (set to 1 or whatever), see beginning of minishell.h file. Since MINISHELL_H is defined, the whole contents of minishell.h is empty (#ifdef doesn't work, we don't have #else, so empty). So include minishell.h in commands.h is substituted with nothing. So contents of commands.h is only its own contents, without anything from minishell.h
- so when the compiler collects minishell.h contents, it pastes commands.h contents in the beginning.
- so when the compiler collects option_env.c contents, it pastes minishell.h (including copy-pasted commands.h) in the beginning
- (you might ask how comes that the struct from minishell.h compiles fine in option_env->minishell.h->commands.h, since there is no text copy-pasted into the beginning of commands.h, so no definitions... and I don't know how to answer this right now. it just works)